llvm/tools/sycl-post-link/CompileTimePropertiesPass.h

```diff
@@ -0,0 +1,108 @@
+//=== CompileTimePropertiesPass.h - SYCL Compile Time Props Pass-*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// A transformation pass which converts values of symbolic attributes
+// (compile-time properties) to integer id-based ones to later map to LLVM IR
+// metadata nodes. The header also contains a number of function templates to
+// extract corresponding attributes and their values.
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "llvm/IR/PassManager.h"
+
+#include <cassert>
+
+namespace llvm {
+
+class CompileTimePropertiesPass
+    : public PassInfoMixin<CompileTimePropertiesPass> {
+public:
+  // Enriches the module with metadata that describes the found variables for
+  // the SPIRV-LLVM Translator.
+  PreservedAnalyses run(Module &M, ModuleAnalysisManager &MAM);
+};
+
+namespace detail {
```

Comment:
It is just a nit, but I think that these details should be moved in a separate utils file.
Maybe an improvement for a separate PR.